internal/manifests/collector/collector.go

```diff
@@ -60,7 +63,18 @@ func Build(params manifests.Params) ([]client.Object, error) {
 			resourceManifests = append(resourceManifests, res)
 		}
 	}
-	routes := Routes(params.Config, params.Log, params.Instance)
+
+	baseDomain := ""
+	if params.Instance.Spec.Ingress.Hostname == "" {
+		domain, err := openshift.GetOpenShiftBaseDomain(context.Background(), params.Client)
+		if err != nil {
+			params.Recorder.Event(&params.Instance, "Error", "Normal", "Failed to get basedomain for Route.")
```

Comment:
Does it make sense to return also an error in that case?

Reply:
hm.. I assume failing early is good. May I ask, whats your thoughts on that @jaronoff97 ?